### PR TITLE
Position floating panel without hacks

### PR DESCRIPTION
**Pull request by kof**
## Description

We have been calculating the position of the floating panel using a relative container in the ancestors.
I was about to add that floating panel to the left side (then it turned out its not needed yet) and diddn't want to add that hack  also to the sidepanel on the left.

Now we don't rely on that parent container to be positioned relatively, but we still need to know what container we position floating panel relative to, so we are using a ref over context.

## Steps for reproduction

1. select an instance
2. typography > click font family value
3. fonts picker should be positioned right next to the right panel

## Code Review

- [ ] hi @istarkov , I need you to do
  - conceptual review architecture

## Before requesting a review

- [x] made a self-review
- [x] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [x] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio-designer/blob/main/apps/designer/docs/test-cases.md) document
- [x] added tests
- [x] if any new env variables are added, added them to `.env.example` and the `designer/env-check.js` if mandatory

### CODE REVIEW

File: apps/designer/app/designer/shared/floating-panel/floating-panel.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+}): [MutableRefObject<HTMLButtonElement | null>, number] => {
+  const triggerRef = useRef<HTMLButtonElement | null>(null);
+  const { container } = useContext(FloatingPanelContext);
+  if (isOpen === false) {
+    // Prevent computation if the panel is closed
+    return [triggerRef, 0];
+  }
+  const containerRect = container?.current?.getBoundingClientRect();
+  const triggerRect = triggerRef.current?.getBoundingClientRect();
+  return [triggerRef, (triggerRect?.left ?? 0) - (containerRect?.left ?? 0)];
```
Comment (TrySound) [inline apps/designer/app/designer/shared/floating-panel/floating-panel.tsx:24]: Accessing dom in render is not a good idea. Refs can still not be set. Use effects

File: apps/designer/app/designer/shared/floating-panel/floating-panel-provider.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { createContext, RefObject } from "react";
+
+export const FloatingPanelContext = createContext<{
```
Comment (TrySound) [inline apps/designer/app/designer/shared/floating-panel/floating-panel-provider.tsx:3]: Do we want to expose context?

Comment (kof) [reply apps/designer/app/designer/shared/floating-panel/floating-panel-provider.tsx:3]: nope, removing from index

File: apps/designer/app/designer/shared/floating-panel/floating-panel.tsx (hunk 1 above)
Comment (kof) [reply apps/designer/app/designer/shared/floating-panel/floating-panel.tsx]: done

**Review by TrySound — APPROVED** [review]

**Pull request merged**